CompressionEngine.java

```diff
@@ -0,0 +1,483 @@
+// CompressionEngine.java
+// Author: Eric Zhuang
+// CodeU Project Group 6
+
+//Converts all aspects of a message into a string format
+//Compresses the message content into a more storage-efficient format
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+
+import codeu.chat.common.Time;
+import codeu.chat.common.Uuid;
+import codeu.chat.common.Uuids;
+
+//TODO: Figure out where to put this in the pipeline
+// Message.java read/write
+// Server.java sendToRelay
+
+public class CompressionEngine{
```

Comment:
Please add javadoc for the class as well for each of the methods. Otherwise, it will be hard to understand what each java method does and the arguments to be passed.